### ISSUE TITLE
Break-outside-of-loop false positives in try block

### ISSUE THREAD

**Issue by ldanilek** [rust-lang/rust-analyzer#14701]
Looks like #14503. has been happening since then and i was waiting for the fix to be published, but it looks like rust-analyzer was released with the fix and my issue persists.

in this function, which notably has a try-block, containing a closure that returns a Result, and within the closure uses the `?` syntax:

```rust
fn _do_thing() {
    let r: anyhow::Result<()> = try {
        let v = vec![1, 2, 3];
        v.iter().map(|i| {
            let result = Some("hi").ok_or(anyhow::anyhow!("bad"));
            result?; // this line has error
            anyhow::Ok(i)
        }).collect::<anyhow::Result<Vec<_>>>()?;
    };
    r.unwrap();
}
```

We get a "break outside of loop" error, which is interesting because the code contains no break statements.

<img width="805" alt="Screen Shot 2023-05-01 at 11 27 22 AM" src="https://user-images.githubusercontent.com/4319355/235506389-c0b5347f-c8b7-4431-a5c2-ccd1b5a9c26a.png">

Note in contrast to #14503, there is nothing async here -- instead we have a try block enabled by `#![feature(try_blocks)]`

**rust-analyzer version**: (eg. output of "rust-analyzer: Show RA Version" command, accessible in VSCode via <kbd>Ctrl/⌘</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>)
happens in pre-release and release which are both `rust-analyzer version: 0.3.1498-standalone (3a27518fe 2023-04-30)`

**rustc version**: (eg. output of `rustc -V`)
`rustc 1.69.0-nightly (31f858d9a 2023-02-28)`

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

**Comment by Veykril** [comment]
Looks like we mess up resetting the tracked try block when we encounter a closure nested in a try block.
